src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
 
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String content = reader.readLine();
+            int totalSupply = 0;
+            int totalBuy = 0;
+            int result = 0;
+            while (content != null) {
+                String[] parts = content.split(",",2);
+                String operation = parts[0].trim();
+
+                if (parts.length != 2) {
+                    continue;
+                }
+
+                int amount = Integer.parseInt(parts[1].trim());
+                content = reader.readLine();
+
+                switch (operation) {
+                    case "supply":
+                        totalSupply += amount;
+                        break;
+                    case "buy":
+                        totalBuy += amount;
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Invalid operation: " + operation);
+                }
+            }
+            result = totalSupply - totalBuy;
+            try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+                writer.write("supply," + totalSupply + System.lineSeparator());
+                writer.write("buy," + totalBuy + System.lineSeparator());
+                writer.write("result," + result + System.lineSeparator());
+            } catch (IOException e) {
+                throw new RuntimeException("Can't create file");
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file");
```

The catch blocks rethrow RuntimeException but neither include the original exception as the cause nor provide a descriptive message that mentions the file. The checklist requires: "Don't ignore exceptions
Leaving an empty catch block or `e.printStackTrace` here is a bad practice. 
Better re-throw `RuntimeException` with the original exception and some message in the parameters:

catch (Exception e) {
    throw new RuntimeException("Can't read data from the file " + fileName, e);
}"
Update both reader and writer catch blocks to rethrow a RuntimeException that includes the original exception (e) and a clear message (preferably including the file name).